pkg/image/validation/kubernetes.go

```diff
@@ -317,9 +323,51 @@ func validateHelmRepoArgs(parsedURL *url.URL, repo *image.HelmRepository) []Fail
 		})
 	}
 
+	if repo.SkipTLSVerify && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'caFile' field for %q should not be defined while 'skipTLSVerify' is true.", repo.Name),
+		})
+	}
+
+	if repo.PlainHTTP && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'caFile' field for %q should not be defined while 'plainHTTP' is true.", repo.Name),
+		})
+	}
+
+	if parsedURL.Scheme == httpScheme && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'url' field for %q contains 'http://' but 'caFile' field is defined.", repo.Name),
+		})
+	}
+
 	return failures
 }
 
+func validateHelmRepoCert(repoName, certFile string, imageConfigDir string) string {
+	if certFile == "" {
+		return ""
+	}
+
+	validExtensions := []string{".pem", ".crt", ".cer", ".der", ".p7b", ".p7c", ".pfx", ".p12"}
+	if !slices.Contains(validExtensions, filepath.Ext(certFile)) {
+		return fmt.Sprintf("Helm chart 'caFile' field for %q must be the name of a valid cert file with the following possible endings: '.pem', '.crt', '.cer', '.der', '.p7b', '.p7c', '.pfx', '.p12'.", repoName)
+	}
+
+	certFilePath := filepath.Join(imageConfigDir, combustion.K8sDir, combustion.HelmDir, combustion.CertsDir, certFile)
+	_, err := os.Stat(certFilePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return fmt.Sprintf("Helm repo cert file '%s' could not be found at '%s'.", certFile, certFilePath)
```

Comment:
I'm not  sure whether mixing up the terminology is the right path here... we're technically only referring to CA bundles and not cert files.